lib/ansible/modules/windows/win_iis_webbinding.ps1

```diff
@@ -52,84 +52,39 @@ function Create-BindingInfo {
 }
 
 # Used instead of get-webbinding to ensure we always return a single binding
+# We can't filter properly with get-webbinding...ex get-webbinding ip * returns all bindings
 # pass it $binding_parameters hashtable
 function Get-SingleWebBinding {
-    $bind_search_splat = @{
-        'name' = $args[0].name
-        'protocol' = $args[0].protocol
-        'port' = $args[0].port
-        'ip' = $args[0].ip
-        'hostheader' = $args[0].hostheader
-    }
 
-    # if no bindings exist, get-webbinding fails with an error that can't be ignored via error actions on older systems
-    # let's ignore that specific error
-    If (-not $bind_search_splat['hostheader'])
-    {
-        Try {
-            Get-WebBinding @bind_search_splat | Where-Object {$_.BindingInformation.Split(':')[-1] -eq [string]::Empty}
-        }
-        Catch {
-            If (-not $_.Exception.Message.CompareTo('Cannot process argument because the value of argument "obj" is null. Change the value of argument "obj" to a non-null value'))
-            {
-                Throw $_.Exception.Message
-            }
-        }
+    Try {
+        $site_bindings = get-webbinding -name $args[0].name
     }
-    Else
-    {
-        Try {
-            Get-WebBinding @bind_search_splat
-        }
-        Catch {
-            If (-not $_.Exception.Message.CompareTo('Cannot process argument because the value of argument "obj" is null. Change the value of argument "obj" to a non-null value'))
-            {
-                Throw $_.Exception.Message
-            }
+    Catch {
+        # 2k8r2 throws this error when you run get-webbinding with no bindings in iis
+        If (-not $_.Exception.Message.CompareTo('Cannot process argument because the value of argument "obj" is null. Change the value of argument "obj" to a non-null value'))
+        {
+            Throw $_.Exception.Message
         }
+        Else { return }
     }
-}
-
-Function Get-CertificateSubjects {
-Param (
-[string]$CertPath
-)
-    If (-Not (Test-Path $CertPath) )
-    {
-        Fail-Json -obj $result -message "Unable to locate certificate at $CertPath"
-    }
-
-    $cert = get-item $CertPath
 
-    If ([version][System.Environment]::OSVersion.Version -ge [version]6.2)
+    Foreach ($binding in $site_bindings)
     {
-        $cert.DnsNameList.unicode
-    }
-    Else
-    {
-        $san = $cert.extensions | Where-Object {$_.Oid.FriendlyName -eq 'Subject Alternative Name'}
-        If ($san)
-        {
-            $san.Format(1) -split '\r\n' | Where-Object {$_} | ForEach-Object {
-                ($_ -split '=')[-1]
-            }
-        }
-        Else
+        $splits = $binding.bindingInformation -split ':'
```

Comment:
Is it guaranteed that BindingInformation always contains 2 `:` in that particular format?

Reply:
Yah. Confirmed on all versions of Windows we support and per the docs:

https://msdn.microsoft.com/en-us/library/microsoft.web.administration.binding.bindinginformation.aspx

If it ever changed though, 1 colon instead of 2, I don't think it would alter the behavior. What's behind the 2nd colon is the host_header value....so without it `$args[0].hostheader -eq $splits[2]` would fail which is the behavior we are looking for anyways in the event that a host header is not defined.